### PR TITLE
LDAP Authentication

### LINKED ISSUE #1: Add LDAP Identity Provider

### PR DESCRIPTION
**Pull request by sabre1041**
Added section on LDAP Authorization to resolve #1

### CODE REVIEW

File: .applier/group_vars/seed-hosts.yml [hunk 1]
```diff
@@ -24,3 +28,15 @@ openshift_cluster_content:
       CLUSTER_ID: "{{ clusterid }}"
       CLOUD_REGION_ZONE: "{{ cloudregion }}c"
     namespace: openshift-machine-api
+- object: LDAP Cluster Authentication
+  content:
+    - name: LDAP Authentication
```
Comment (jkupferer) [inline .applier/group_vars/seed-hosts.yml]: Let's keep consistent indentation style in the file, looks like no indent before lists?

File: .applier/group_vars/seed-hosts.yml [hunk 2]
```diff
@@ -24,3 +28,15 @@ openshift_cluster_content:
...
+      template: "{{ inventory_dir }}/../templates/ldap-auth.yaml"
+      params_from_vars:
+        LDAP_CA: "{{ ldap_ca }}"
+        LDAP_BIND_PASSWORD: "{{ ldap_bind_password }}"
+        LDAP_BIND_DN: "{{ ldap_bind_dn }}"
+        LDAP_SEARCH_URL: "{{ ldap_search_url }}"
+      namespace: openshift-config
+      tags:
+        - ldap_auth
```
Comment (jkupferer) [inline .applier/group_vars/seed-hosts.yml]: Same thing here, let's keep consistent indent. Do we have yamllint?

File: README.md [hunk 3]
```diff
@@ -36,3 +36,38 @@ OK, now let's roll out our first config. We're going to do this using OpenShift
...
+
+_Note: THe use of the `filter_tags` variable allows for a subset of the Applier inventory to be executed._
+
+With the configurations applied, attempt to login with a user defined in the LDAP instance
+
+    oc login -u <user> -p <password>
+
+To login to the web console, locate the URL of the OpenShift web console and navigate to the URL presented
+
+    oc get routes -n openshift-console console -o jsonpath="https://{.spec.host}"
```
Comment (jkupferer) [inline README.md]: Better command is `oc whoami --show-console`

File: templates/ldap-auth.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    required: true
+  - name: LDAP_BIND_PASSWORD
+    description: LDAP Bind password
+    required: true
+  - name: LDAP_BIND_DN
+    description: LDAP Bind Distinguished Name
+    required: true
+  - name: LDAP_SEARCH_URL
+    description: LDAP Search URL
+    required: true
```
Comment (jkupferer) [inline templates/ldap-auth.yaml]: Missing newline at end of file. Also would prefer consistent YAML indent style.

File: README.md (hunk 3 above)
Comment (sabre1041) [reply README.md]: @jkupferer this is not a valid command

**Comment by sabre1041** [comment]
@jkupferer rebased with yamllint changes and proposed formatting changes

File: README.md (hunk 3 above)
Comment (jkupferer) [reply README.md]: What oc version are you using?
```
$ oc whoami --show-console
https://master.dev311.openshift.opentlc.com:443
$ oc version
Client Version: version.Info{Major:"4", Minor:"1+", GitVersion:"v4.1.0", GitCommit:"159c28d85", GitTreeState:"clean", BuildDate:"2019-05-03T10:44:21Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0+d4cacc0", GitCommit:"d4cacc0", GitTreeState:"clean", BuildDate:"2019-02-08T01:20:07Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by sabre1041** [comment]
@jkupferer only the very latest CLI has the `oc whoami` command. Updated the README

**Pull request merged**